build/codegen/codegen.sh

```diff
@@ -17,15 +17,24 @@
 # shellcheck disable=SC2086,SC2089,SC2090
 # Disables quote checks, which is needed because of the SED variable here.
 
+KUBE_CODE_GEN_VERSION="kubernetes-1.17.2"
+GROUP_VERSIONS="rook.io:v1 rook.io:v1alpha2 ceph.rook.io:v1 cockroachdb.rook.io:v1alpha1 nfs.rook.io:v1alpha1 cassandra.rook.io:v1alpha1 edgefs.rook.io:v1 yugabytedb.rook.io:v1alpha1"
+
 scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+codegendir="${scriptdir}/../../vendor/k8s.io/code-generator"
+
+# vendoring k8s.io/code-generator temporarily
+echo "require k8s.io/code-generator ${KUBE_CODE_GEN_VERSION}" >> ${scriptdir}/../../go.mod
+go mod vendor
+git checkout HEAD ${scriptdir}/../../go.mod ${scriptdir}/../../go.sum
 
-cd "${scriptdir}/../../vendor/k8s.io/code-generator" && ./generate-groups.sh \
+bash ${codegendir}/generate-groups.sh \
     all \
     github.com/rook/rook/pkg/client \
     github.com/rook/rook/pkg/apis \
-    "rook.io:v1 rook.io:v1alpha2 ceph.rook.io:v1 cockroachdb.rook.io:v1alpha1 nfs.rook.io:v1alpha1 cassandra.rook.io:v1alpha1 edgefs.rook.io:v1 yugabytedb.rook.io:v1alpha1"
-# this seems busted in the release-1.8 branch
-#  --go-header-file ${SCRIPT_ROOT}/build/codegen/header.txt
+    "${GROUP_VERSIONS}"
+# cleanup vendor
+rm -fr ${scriptdir}/../../vendor
```

Comment:
Why delete the entire vendor folder? Seems like we should leave it there.

Reply:
I think the vendor folder is used only at executing `make codegen` and it's not necessary to leave it. (right?) So I delete it at the end of the process. If it's better to leave the vendor folder for other reasons , I'll fix it!

Reply:
How about let's leave it there for now. If there is a more scoped dir you want to cleanup, it wouldn't hurt though.

Reply:
I understand.